src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -629,29 +633,30 @@ public bool IsFrozen
         {
             get
             {
-                return IsAnyUpstreamFrozen() || isFrozenExplicitly;               
+                return IsAnyUpstreamFrozen() || isFrozenExplicitly;
             }
             set
             {
-                isFrozenExplicitly = value;                    
+                isFrozenExplicitly = value;
                 //If the node is Unfreezed then Mark all the downstream nodes as
                 // modified. This is essential recompiling the AST.
                 if (!value)
-                {                   
-                    MarkDownStreamNodesAsModified(this);                    
-                    OnNodeModified();
+                { 
+                    MarkDownStreamNodesAsModified(this);
+                    OnNodeModified();                   
                 }
                 //If the node is frozen, then do not execute the graph immediately.
                 // delete the node and its downstream nodes from AST.
                 else
                 {
-                    OnUpdateASTCollection();
-                }                   
+                    ComputeUpstreamNodes();
```

Comment:
If a node is explicitly frozen, then compute the downstream nodes from that node.